File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/video/FallbackStrategy.kt
package dev.hebei.camerax_android.video

class FallbackStrategy internal constructor(internal val obj: androidx.camera.video.FallbackStrategy) {
    companion object {
        fun higherQualityOrLowerThan(quality: Quality): FallbackStrategy {
            val obj = androidx.camera.video.FallbackStrategy.higherQualityOrLowerThan(quality.obj)
            return FallbackStrategy(obj)
        }

        fun higherQualityThan(quality: Quality): FallbackStrategy {
            val obj = androidx.camera.video.FallbackStrategy.higherQualityThan(quality.obj)
            return FallbackStrategy(obj)
        }

        fun lowerQualityOrHigherThan(quality: Quality): FallbackStrategy {
            val obj = androidx.camera.video.FallbackStrategy.lowerQualityOrHigherThan(quality.obj)
            return FallbackStrategy(obj)
        }

        fun lowerQualityThan(quality: Quality): FallbackStrategy {
            val obj = androidx.camera.video.FallbackStrategy.lowerQualityThan(quality.obj)
            return FallbackStrategy(obj)
        }
    }
}

File: camerax/lib/src/video/fallback_strategy.dart
import 'package:camerax_platform_interface/camerax_platform_interface.dart'
    as $interface;
import 'package:meta/meta.dart';

final class FallbackStrategy {
  final $interface.FallbackStrategy _obj;

  FallbackStrategy.higherQualityOrLowerThan($interface.Quality quality)
      : _obj = $interface.FallbackStrategy.higherQualityOrLowerThan(quality);

  FallbackStrategy.higherQualityThan($interface.Quality quality)
      : _obj = $interface.FallbackStrategy.higherQualityThan(quality);

  FallbackStrategy.lowerQualityOrHigherThan($interface.Quality quality)
      : _obj = $interface.FallbackStrategy.lowerQualityOrHigherThan(quality);

  FallbackStrategy.lowerQualityThan($interface.Quality quality)
      : _obj = $interface.FallbackStrategy.lowerQualityThan(quality);

  @internal
  $interface.FallbackStrategy get obj => _obj;
}
